### PR TITLE
Fix daid STOP LCD behavior

### LINKED ISSUE #2613: Sub-issue (2602): Fix daid STOP LCD behavior
## Summary

Fix daid STOP instruction LCD behavior for DMG and CGB non-Mode-3 cases.

## Problem

Issue #2602 added daid suite automation scaffolding and a CPU STOP state so STOP no longer resumes from ordinary VBlank interrupts. The CGB Mode 3 STOP case now matches the upstream reference, but the normal stop_instr cases still display the failure text instead of matching the blank upstream references.

Observed with 500-frame captures:

- stop_instr_gbc_mode3_cgbe matches upstream stop_instr_gbc_mode3.png with CRC 0x4D1E1139.
- stop_instr_dmg0 produces CRC 0x7D7595AD and DMG-A/B/C produce CRC 0x6763B6A5, while upstream stop_instr.dmg.png decodes to CRC 0x811BB2FB.
- stop_instr_cgbe produces CRC 0x7D7595AD, while upstream stop_instr.gbc.png decodes to CRC 0x252F710F.

Pan Docs says STOP is terminated by joypad lines going low. It also says CGB STOP with LCD enabled results in a black screen except in Mode 3, where it keeps drawing the current screen.

## Scope

- Implement the missing STOP LCD/PPU side effects needed by daid stop_instr on DMG and CGB.
- Preserve the passing CGB Mode 3 behavior.
- Activate the ignored stop_instr DMG and CGB daid tests.
- Keep or extend focused CPU/PPU tests for STOP state behavior.

## Out of scope

- CGB speed-switch LY/STAT timing.
- Mid-scanline BGP timing.
- Full modelling of every undocumented STOP edge case beyond the daid-visible behavior.

## Acceptance criteria

- stop_instr tests for DMG0, DMG-A, DMG-B, DMG-C, and CGB-E are active, non-ignored tests.
- stop_instr_gbc_mode3_cgbe remains active and passing.
- The normal STOP cases match the upstream blank reference images after visual review.
- Existing HALT and CGB speed-switch tests are not regressed.

## Validation

cargo test --no-default-features --lib gb::integration_tests::daid_tests::test_stop_instr -- --test-threads=1

cargo test --no-default-features --lib gb::cpu::sm83::tests::test_stop -- --test-threads=1

NESER_CAPTURE_SCREEN=1 cargo test --no-default-features --lib gb::integration_tests::daid_tests::capture_all_daid_screenshots -- --ignored --nocapture

Compare the generated stop_instr captures in target/daid-captures against the upstream PNGs in roms/gb/automated_tests/daid.

## Dependencies / Links

Parent: #2602

### PR DESCRIPTION
**Pull request by rmstdope**
## Summary

Fixes #2613.

- Models normal STOP LCD output with a PPU display override:
  - DMG STOP blanks to the white daid reference.
  - CGB non-Mode-3 STOP blanks to the black daid/Pan Docs reference.
  - CGB Mode 3 STOP preserves the current framebuffer.
- Keeps PPU frame cadence running while STOP display output is overridden, so frame-based tests and frontends do not hang.
- Activates the daid `stop_instr` DMG/CGB regression tests and updates them to the upstream reviewed CRCs.
- Reconciles STOP display state after save-state load for older stopped CPU snapshots without overwriting newly serialized CGB Mode 3 preserve state.

## Validation

- `cargo clippy --all-targets --all-features -- -D warnings`
- `cargo fmt --check`
- `./scripts/test-dir.sh src/gb --skip-integration`
- `cargo test --no-default-features --lib`
- `wasm-pack test --headless --chrome --no-default-features --features wasm`
- `source .venv/bin/activate && python -m unittest discover -s scripts -t scripts -p "test_*.py" && python -m unittest discover -s scripts/metadata-scraper -t scripts/metadata-scraper -p "test_*.py" && python -m unittest discover -s scripts/nes-rom-db-scraper -t scripts/nes-rom-db-scraper -p "test_*.py"`
- `npm test`
- `NESER_CAPTURE_SCREEN=1 cargo test --no-default-features --lib gb::integration_tests::daid_tests::capture_all_daid_screenshots -- --ignored --nocapture`

STOP capture CRCs:

- `stop_instr_dmg0`, `stop_instr_dmga`, `stop_instr_dmgb`, `stop_instr_dmgc`: `0x811BB2FB`
- `stop_instr_cgbe`: `0x252F710F`
- `stop_instr_gbc_mode3_cgbe`: `0x4D1E1139`

### CODE REVIEW

File: src/gb/ppu/ppu.rs [hunk 1]
```diff
@@ -322,6 +351,10 @@ impl Ppu {
     }
 
     fn render_pixel_fifo_dot(&mut self) {
+        if self.stop_display_mode != StopDisplayMode::Inactive {
+            return;
+        }
+
```
[the remark this answers is not in the document]
Comment (rmstdope) [reply src/gb/ppu/ppu.rs]: Addressed in `bca0a7ee`: STOP display suppression now still calls `PixelFifoRenderer::tick` and passes a `suppress_output` flag so FIFO/window state advances normally while framebuffer writes are skipped. I also added `test_stop_display_suppresses_pixels_but_keeps_fifo_advancing` to cover the mid-scanline behavior.

File: src/gb/ppu/screen_buffer.rs [hunk 2]
```diff
@@ -99,6 +107,24 @@ mod tests {
...
+
+        // When: the buffer is filled
+        buf.fill_rgb(0xAA, 0xBB, 0xCC);
+
+        // Then: every sampled pixel has the fill colour
+        assert_eq!(buf.get_pixel(0, 0), (0xAA, 0xBB, 0xCC));
+        assert_eq!(
+            buf.get_pixel(ScreenBuffer::WIDTH - 1, ScreenBuffer::HEIGHT - 1),
+            (0xAA, 0xBB, 0xCC)
+        );
```
[the remark this answers is not in the document]
Comment (rmstdope) [reply src/gb/ppu/screen_buffer.rs:125]: Addressed in `bca0a7ee`: `test_fill_rgb_sets_entire_buffer` now validates every pixel chunk in the full framebuffer snapshot instead of sampling only first/last pixels.
